gulpfile.js

```diff
@@ -48,6 +50,46 @@ function watch(){
     webpack(webpackConfig).watch(300, webpackOnBuild());
 }
 
+function watchSharedPortalComponents(done){
+
+    let sharedComponentPath = path.resolve(__dirname, 'node_modules/@hms-dbmi-bgm/shared-portal-components');
+    let isLinked = false;
+
+    try { // Get exact path to dir, else leave. Used to avoid needing to webpack dependency itself.
+        for (var i = 0; i < 10; i++) { // Incase multiple links.
+            sharedComponentPath = fs.readlinkSync(sharedComponentPath);
+            isLinked = true;
+        }
+    } catch (e){
+        // ... not linked
+    }
+
+    console.log(
+        "`@hms-dbmi-bgm/shared-portal-components` directory is",
+        isLinked ? "sym-linked to `" + sharedComponentPath + "`." : "NOT sym-linked."
+    );
+
+    if (!isLinked){ // Exit
+        done();
+    }
+
+    // Same as shared-portal-components own build method, but with "--watch"
```

Comment:
Also uses 4DN's `@babel/cli` (and config) instead of shared-portal-components, but this shouldn't mater as should be equal.

To be safe & future-proof, and not instal babel/cli in 4DN, perhaps could run something like `sharedComponentPath/node_modules/.bin/babl` instead of `npx babel` ... maybe.